File: design-patterns/go/facade/travel/weather.go
// weather.go

package main

import "fmt"

type Weather struct {
}

func NewWeather() (weather *Weather) {
	weather = &Weather{}
	return
}

func (weather *Weather) GetWeatherInfo(lat float64, lng float64) {
	fmt.Println("Temperature: 20.7")
	fmt.Println("Precipitation: 1%")
	fmt.Println("Humidity: 73%")
	fmt.Println("Wind: 8 km/h")
}

File: design-patterns/ts/facade/travel/weather.ts
// weather.ts

class Weather {
    getWeatherInfo(lat: number, lng: number): void {
        console.log("Temperature: 20.7");
        console.log("Precipitation: 1%");
        console.log("Humidity: 73%");
        console.log("Wind: 8 km/h");
    }
}

export default Weather;